src/System.Data.SqlClient/src/Interop/LocaleInterop.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace System.Data
+{
+    internal static class LocaleInterop
+    {
+        // Maps between LCIDs and LocaleName+AnsiCodePages+Encodings
+        private static readonly ConcurrentDictionary<int, Tuple<string, int, Encoding>> s_cachedEncodings = new ConcurrentDictionary<int, Tuple<string, int, Encoding>>();
+
+        private const string ApiWinCoreLocalization = "api-ms-win-core-localization-l1-2-0.dll";
```

Comment:
What reason not to follow [Interop Guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md) :question: